cleanlab/datalab/internal/issue_manager/data_valuation.py

```diff
@@ -142,26 +140,22 @@ def find_issues(
                 )
             if self.metric is None:
                 self.metric = "cosine" if features.shape[1] > 3 else "euclidean"
-            knn = NearestNeighbors(n_neighbors=self.k, metric=self.metric).fit(features)
+            knn_graph = None
 
-            if self.metric and self.metric != knn.metric:
-                warnings.warn(
-                    f"Metric {self.metric} does not match metric {knn.metric} used to fit knn. "
-                    "Most likely an existing NearestNeighbors object was passed in, but a different "
-                    "metric was specified."
-                )
-            self.metric = knn.metric
-
-            try:
-                check_is_fitted(knn)
-            except NotFittedError:
-                knn.fit(features)
-
-            knn_graph = knn.kneighbors_graph(mode="distance")
```

Comment:
Please revert to this logic on deciding whether to compute a `knn_graph` within Datalab (and this issue manager).
The cleanlab.data_valuation module should not be responsible for computing a knn_graph **and passing it** on to Datalab.
It's fine to copy the helper function to the data_valuation module in the case when features are passed to the function in the `cleanlab.data_valuation`module.